packages/flutter_test/lib/src/window.dart

```diff
@@ -944,3 +982,263 @@ class TestPlatformDispatcher implements ui.PlatformDispatcher {
     return null;
   }
 }
+
+/// [ui.FlutterView] that wraps another [ui.FlutterView] and allows faking of
+/// some properties for testing purposes.
+///
+/// See also:
+///
+///   * [TestPlatformDispatcher], which allows for faking of platform specific
+///     functionality.
+class TestFlutterView implements ui.FlutterView {
+  /// Constructs a [TestFlutterView] that defers all behavior to the given
+  /// [ui.FlutterView] unless explicitly overridden for testing.
+  TestFlutterView({
+    required ui.FlutterView view,
+    required TestPlatformDispatcher platformDispatcher,
+  }) : _view = view, _platformDispatcher = platformDispatcher {
+    _viewConfiguration = view.viewConfiguration.copyWith(view: this);
+  }
+
+  /// The [ui.FlutterView] backing this [TestFlutterView].
+  final ui.FlutterView _view;
+
+  final TestPlatformDispatcher _platformDispatcher;
+  late ui.ViewConfiguration _viewConfiguration;
+
+  @Deprecated(
+    'Use _viewConfiguration instead. '
+    'This is only available to enable backwards compatibility of TestWindow. '
+    'This feature was deprecated after v3.8.0.10.1.pre.'
+  )
+  ui.ViewConfiguration? _overrideConfiguration;
+
+  @override
+  TestPlatformDispatcher get platformDispatcher => _platformDispatcher;
+
+  @override
+  ui.ViewConfiguration get viewConfiguration => _overrideConfiguration ?? _viewConfiguration;
+
+  @Deprecated(
+    'Use setters for individual test properties instead. '
+    'This is only available to enable backwards compatibility of TestWindow. '
+    'This feature was deprecated after v3.8.0.10.1.pre.'
+  )
+  /// Sets the ViewConfiguration directly. Not supported publicly by
+  /// [TestFlutterView] to avoid ambiguity between setting the configuration
+  /// directly and using the test setters. Required temporarily for backwards
+  /// compatibility of [TestWindow].
+  // ignore: avoid_setters_without_getters
+  set _viewConfigurationLegacy(ui.ViewConfiguration? viewConfiguration) {
+    if (viewConfiguration == null) {
+      _overrideConfiguration = null;
+    } else {
+      _overrideConfiguration = viewConfiguration.copyWith(view: this);
+    }
+    platformDispatcher.onMetricsChanged?.call();
+  }
+
+  @override
+  Object get viewId => _view.viewId;
+
+  @override
```

Comment:
here and elsewhere: the overrides should come after the doc comment.

Reply:
I think I got all of them.